### PR TITLE
[RFC] Windows: Implement os_setenv()

### PR DESCRIPTION
**Pull request by sethjackson**
Cherry picked from #810. From equalsraf@f12db3c.

### CODE REVIEW

**Comment by sethjackson** [comment]
Not really sure why QuickBuild is failing here...

File: config/config.h.in [hunk 1]
```diff
@@ -30,6 +30,7 @@
 #cmakedefine HAVE_NL_LANGINFO_CODESET
 #cmakedefine HAVE_NL_MSG_CAT_CNTR
 #cmakedefine HAVE_PUTENV
+#cmakedefine HAVE_PUTENV_S
```
Comment (justinmk) [inline config/config.h.in]: `HAVE_PUTENV` is not actually used anywhere, so it could be removed from here and `config/CMakeLists.txt`.

**Comment by sethjackson** [comment]
@justinmk I removed `HAVE_PUTENV from `config/config.h.in` but I couldn't find it in `config/CMakeLists.txt`. Maybe I missed something?

**Comment by justinmk** [comment]
Right, you removed it already in this PR :) Thanks

**Pull request merged**